### PR TITLE
Throw exception when constructing Value with NaN or infinite numbers.

### PR DESCRIPTION
**Pull request by pengdev**
### Summary of changes

Throw exception when constructing `Value` with not a finite number, so we get meaningful crash stacktrace early.

According to [JSON](https://www.json.org/json-en.html)  NAN isn't part of the JSON specification, and it will cause issues when we pass the Double.NaN to gl-native, where we serialise it to a JSON string.

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [x] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: extension-style/src/main/java/com/mapbox/maps/extension/style/utils/TypeUtils.kt [hunk 1]
```diff
@@ -44,6 +44,7 @@ internal object TypeUtils {
         Value(value)
       }
       is Double -> {
+        require(!value.isNaN()) { "Value can not be Double.NaN" }
```
Comment (jush) [inline extension-style/src/main/java/com/mapbox/maps/extension/style/utils/TypeUtils.kt:47]: Should we do the same for `Float` below? and their array counterparts?

**Review by jush — CHANGES_REQUESTED** [review]

File: extension-style/src/main/java/com/mapbox/maps/extension/style/utils/TypeUtils.kt (hunk 1 above)
Comment (jush) [reply extension-style/src/main/java/com/mapbox/maps/extension/style/utils/TypeUtils.kt:47]: Actually, the same will happen if INFINITY is passed.
See Moshi tests: https://github.com/square/moshi/blob/master/moshi/src/test/java/com/squareup/moshi/MoshiTest.java#L296

Comment (pengdev) [reply extension-style/src/main/java/com/mapbox/maps/extension/style/utils/TypeUtils.kt]: Fixed, please take a look again.

**Review by jush — APPROVED** [review]

**Review by kiryldz — APPROVED** [review]

File: extension-style/src/main/java/com/mapbox/maps/extension/style/utils/TypeUtils.kt [hunk 2]
```diff
@@ -44,32 +44,34 @@ internal object TypeUtils {
...
       is Float -> {
+        require(value.isFinite()) { "Value can not be Float.NaN, Float.POSITIVE_INFINITY or Float.NEGATIVE_INFINITY" }
         Value(value.toDouble())
       }
       is Long -> {
         Value(value)
       }
       is IntArray -> {
-        val valueArray = value.map { Value(it.toLong()) }
+        val valueArray = value.map { wrapToValue(it.toLong()) }
```
Comment (yunikkk) [inline extension-style/src/main/java/com/mapbox/maps/extension/style/utils/TypeUtils.kt:58]: Why do you need wrap int though?

File: extension-style/src/test/java/com/mapbox/maps/extension/style/utils/TypeUtilsTest.kt [hunk 3]
```diff
@@ -58,12 +58,42 @@ class TypeUtilsTest {
...
+  fun wrapToValue_Float_POSITIVE_INFINITY() {
+    testwrapToValue(Float.POSITIVE_INFINITY)
+  }
+
+  @Test(expected = IllegalArgumentException::class)
+  fun wrapToValue_Float_NEGATIVE_INFINITY() {
+    testwrapToValue(Float.NEGATIVE_INFINITY)
+  }
+
   private fun testwrapToValue(value: Any) {
```
Comment (yunikkk) [inline extension-style/src/test/java/com/mapbox/maps/extension/style/utils/TypeUtilsTest.kt:97]:
```suggestion
  private fun testWrapToValue(value: Any) {
```

File: extension-style/src/main/java/com/mapbox/maps/extension/style/utils/TypeUtils.kt (hunk 2 above)
Comment (pengdev) [reply extension-style/src/main/java/com/mapbox/maps/extension/style/utils/TypeUtils.kt:58]: just want to make it consistent, so we can add future assertion to the primitive types as well.

**Review by yunikkk — APPROVED** [review]

**Pull request merged**